plugins/inputs/gcp_stackdriver/gcp_stackdriver.go

```diff
@@ -0,0 +1,473 @@
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		metricType := metricDescriptor.Type
+		valueType := metricDescriptor.ValueType
+
+		if s.includeMetricType(metricType) {
+			if valueType == metricpb.MetricDescriptor_DISTRIBUTION {
+				if s.ScrapeDistributionBuckets {
+					tsConf := s.newTimeSeriesConf(metricType)
+					tsConf = tsConf.initForDistribution()
+					ret = append(ret, *tsConf)
+				}
+				for _, alignerStr := range s.DistributionAggregationAligners {
+					tsConf := s.newTimeSeriesConf(metricType)
+					tsConf = tsConf.initForAggregate(alignerStr)
+					ret = append(ret, *tsConf)
+				}
+			} else {
+				ret = append(ret, *s.newTimeSeriesConf(metricType))
+			}
+		}
+	}
+
+	return ret, nil
+}
+
+// Do the work! Create a (rate-limited) goroutine for each timeSeriesConf. The
+// rate limiting is super-necessary: GCP will trout-slap us if we talk to it
+// too quickly.
+func (s *GCPStackdriver) scrapeAllTimeSeries(
+	acc telegraf.Accumulator,
+	types []timeSeriesConf) error {
+
+	wg := &sync.WaitGroup{}
+	lmtr := limiter.NewRateLimiter(s.RateLimit, time.Second)
+
+	defer lmtr.Stop()
+	wg.Add(len(types))
+	for _, timeSeriesRequest := range types {
+		go func(tsc timeSeriesConf) {
+			<-lmtr.C
+			defer wg.Done()
+			acc.AddError(s.scrapeTimeSeries(acc, tsc))
+		}(timeSeriesRequest)
+	}
+
+	wg.Wait()
+	return nil
+}
+
+// Do the work to scrape an individual time series. Runs inside a
+// timeseries-specific goroutine.
+func (s *GCPStackdriver) scrapeTimeSeries(acc telegraf.Accumulator,
+	tsConf timeSeriesConf) error {
+
+	tsReq := tsConf.listTimeSeriesRequest
+	measurement := tsConf.measurement
+	fieldPrefix := tsConf.fieldPrefix
+	tsResp := s.client.ListTimeSeries(context.TODO(), &tsReq)
+
+	for {
+		tsDesc, tsErr := tsResp.Next()
+		if tsErr == iterator.Done {
+			break
+		}
+		if tsErr != nil {
+			log.Printf("Request %s failure: %s\n", tsReq.String(), tsErr)
+			return tsErr
+		}
+
+		tags := map[string]string{
+			"resource_type": tsDesc.Resource.Type,
+		}
+		for k, v := range tsDesc.Resource.Labels {
+			if s.includeTag(k) {
+				tags[k] = v
+			}
+		}
+		for k, v := range tsDesc.Metric.Labels {
+			if s.includeTag(k) {
+				tags[k] = v
+			}
+		}
+		for _, p := range tsDesc.Points {
+			ts := time.Unix(p.Interval.EndTime.Seconds, 0)
+
+			var fields map[string]interface{}
+
+			if tsDesc.ValueType == metricpb.MetricDescriptor_DISTRIBUTION {
+				val := p.Value.GetDistributionValue()
+				fields = s.scrapeDistribution(acc, measurement, fieldPrefix, val)
+			} else {
+				var field interface{}
+
+				switch tsDesc.ValueType {
+				case metricpb.MetricDescriptor_BOOL:
+					field = p.Value.GetBoolValue()
+				case metricpb.MetricDescriptor_INT64:
+					field = p.Value.GetInt64Value()
+				case metricpb.MetricDescriptor_DOUBLE:
+					field = p.Value.GetDoubleValue()
+				case metricpb.MetricDescriptor_STRING:
+					field = p.Value.GetStringValue()
+				default:
+					// TODO: AddError here? Is it really an error?
```

Comment:
Maybe either skip the metric, as you currently are, or default it to a chosen type.